VideoLocker/src/main/java/org/edx/mobile/task/SetThreadReadTask.java

```diff
@@ -11,22 +11,15 @@
 
 public class SetThreadReadTask extends Task<DiscussionThread> {
     private final DiscussionThread thread;
-    private final boolean read;
 
     public SetThreadReadTask(@NonNull Context context,
-                             @NonNull DiscussionThread thread, boolean read) {
+                             @NonNull DiscussionThread thread) {
         super(context);
         this.thread = thread;
-        this.read = read;
     }
 
     @Override
     public DiscussionThread call() throws RetroHttpException {
-        return environment.getDiscussionAPI().setThreadRead(thread, read);
-    }
-
-    @Override
-    protected void onSuccess(DiscussionThread discussionThread) {
-        EventBus.getDefault().post(new DiscussionThreadUpdatedEvent(discussionThread));
+        return environment.getDiscussionAPI().setThreadRead(thread, true);
```

Comment:
> But it is getting the thread too, and in fact that's the main purpose of it as far as the client code is concerned (marking it as read being a useful side-effect of the implementation).

Since I am keeping the GET call for thread details, it doesn't make any sense in having 2 different calls having the same name.